### ISSUE TITLE
[BUG] StringFormatter doesn't apply nan_format

### ISSUE THREAD

**Issue by maximlt** [bokeh/bokeh#12172]
### Software versions

Python version      :  3.7.12 | packaged by conda-forge | (default, Oct 26 2021, 06:08:21) 
IPython version     :  7.33.0
Tornado version     :  6.1
Bokeh version       :  2.4.3
BokehJS static path :  /home/liquetm/miniconda3/envs/panel-dev37/lib/python3.7/site-packages/bokeh/server/static
node.js version     :  v17.9.0
npm version         :  8.5.5
Operating system    :  Linux-5.4.0-117-generic-x86_64-with-debian-buster-sid

### Expected behavior

The [documentation](https://docs.bokeh.org/en/latest/docs/reference/models/widgets.tables.html#bokeh.models.widgets.tables.ScientificFormatter.nan_format) of the `StringFormatter` indicates that `nan_format` can be used to format `None` and `NaN` values in a column. However as shown in the example and the screenshot this doesn't seem to be applied at all.

### Observed behavior

It seems like the `StringFormatter` code doesn't handle `nan_format` at all, but I may have missed another place where the formatter is handled:

https://github.com/bokeh/bokeh/blob/423ba41ea57c08cb37021e8ed5a0fb6807c6e9b2/bokehjs/src/lib/models/widgets/tables/cell_formatters.ts#L69

Happy to work on a PR if this is an oversight.

### Example code

```Python
from bokeh.plotting import curdoc

from bokeh.models import ColumnDataSource, DataTable, TableColumn, StringFormatter
import numpy as np

s = ['a'] * 5
s1 = s.copy()
s1[0] = None
s2 = s.copy()
s2[0] = np.nan

data = dict(s1=s1, s2=s2)
source = ColumnDataSource(data)

columns = [
        TableColumn(field="s1", formatter=StringFormatter(text_color='red', nan_format='-')),
        TableColumn(field="s2", formatter=StringFormatter(text_color='red', nan_format='-')),
    ]
data_table = DataTable(source=source, columns=columns, width=400, height=280)

curdoc().add_root(data_table)
```

### Screenshots

![image](https://user-images.githubusercontent.com/35924738/173511361-40720a80-458d-4594-981a-bccee0ecd41d.png)

**Comment by bryevdv** [comment]
`StringFormatter` does not do anything at all with `nan_format` but I am also not sure that it should. The assumption for `StringFormatter` is that the inputs are all already strings to be presented as-is. For this basic formatter, and nan/null/none formatting should be done up front. 

 I am inclined to just remove this property altogether for `branch-3.0` cc @bokeh/dev

**Comment by maximlt** [comment]
Thanks for your answer! For what it's worth, as a user, I would actually find quite useful to have `nan_format` implemented for the `StringFormatter`. And for context, I opened this issue while I was trying someone would was using the `StringFormatter` to format a column in a Panel Tabulator widget, that person had already used `nan_format` with the `NumberFormatter` and expected the same option to work with the `StringFormatter`.

For the sake of completeness I'm adding to the reproducible example a column created from a Pandas `StringDtype` series that holds a `pd.NA` value. With Pandas apparently improving its string support, I'd expect to see more and more usage of this new dtype and `pd.NA`.

```python
from bokeh.plotting import curdoc

from bokeh.models import ColumnDataSource, DataTable, TableColumn, StringFormatter
import numpy as np

s = ['a'] * 5
s1 = s.copy()
s1[0] = None
s2 = s.copy()
s2[0] = np.nan
s3 = pd.Series(s2).astype('string')

data = dict(s1=s1, s2=s2, s3=s3)
source = ColumnDataSource(data)

columns = [
        TableColumn(field="s1", formatter=StringFormatter(text_color='red', nan_format='-')),
        TableColumn(field="s2", formatter=StringFormatter(text_color='red', nan_format='-')),
        TableColumn(field="s3", formatter=StringFormatter(text_color='red', nan_format='-')),
    ]
data_table = DataTable(source=source, columns=columns, width=400, height=280)

curdoc().add_root(data_table)
```

<img width="437" alt="image" src="https://user-images.githubusercontent.com/35924738/175533095-84e4b093-e0cb-42d3-98bd-f06a9b2957b8.png">

**Comment by jbednar** [comment]
I agree with @maximlt that supporting nan_format for strings makes good sense and does seem like it will become of greater importance given how Pandas is going...

**Comment by bryevdv** [comment]
I think this is a larger question. If `pd.NA` moves out of experimental status then we need to figure out how to serialize it properly as a "missing value" as it is intended. I don't think we should abuse `NaN` for this purpose in non-numeric arrays. Then we would want a separate new `missing_format` (or `na_format` though that is typo-close to `nan_format`)

**Comment by mattpap** [comment]
> If pd.NA moves out of experimental status then we need to figure out how to serialize it properly (...)

This is easily possible with the new serializer. In particular, we can have `NA` JS symbol in bokehjs for representing this state. Symbols are unique from everything else, which is an effective way for implementing singleton values.
